integration/aggregate_stats_compat_test.go

```diff
@@ -0,0 +1,131 @@
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/mongo-driver/bson"
+
+	"github.com/FerretDB/FerretDB/integration/setup"
+	"github.com/FerretDB/FerretDB/integration/shareddata"
+)
+
+func TestAggregateCompatCollStats(t *testing.T) {
+	t.Parallel()
+
+	for name, tc := range map[string]struct {
+		skip       string                   // skip test for all handlers, must have issue number mentioned
+		collStats  bson.D                   // required
+		resultType compatTestCaseResultType // defaults to nonEmptyResult
+	}{
+		"NilCollStats": {
+			collStats:  nil,
+			resultType: emptyResult,
+		},
+		"EmptyCollStats": {
+			collStats: bson.D{},
+		},
+		"Count": {
+			collStats: bson.D{{"count", bson.D{}}},
+		},
+		"StorageStats": {
+			collStats: bson.D{{"storageStats", bson.D{}}},
+		},
+		"StorageStatsWithScale": {
+			collStats: bson.D{{"storageStats", bson.D{{"scale", 1000}}}},
+		},
+		"CountAndStorageStats": {
+			collStats: bson.D{{"count", bson.D{}}, {"storageStats", bson.D{}}},
+		},
+	} {
+		name, tc := name, tc
+		t.Run(name, func(t *testing.T) {
+			if tc.skip != "" {
+				t.Skip(tc.skip)
+			}
+
+			t.Helper()
+			t.Parallel()
+
+			// It's enough to use a couple of providers: one for some collection and one for a non-existent collection.
+			s := setup.SetupCompatWithOpts(t, &setup.SetupCompatOpts{
+				Providers:                []shareddata.Provider{shareddata.ArrayDocuments},
+				AddNonExistentCollection: true,
+			})
+			ctx, targetCollections, compatCollections := s.Ctx, s.TargetCollections, s.CompatCollections
+
+			var nonEmptyResults bool
+			for i := range targetCollections {
+				targetCollection := targetCollections[i]
+				compatCollection := compatCollections[i]
+
+				t.Run(targetCollection.Name(), func(t *testing.T) {
+					t.Helper()
+
+					command := bson.A{bson.D{{"$collStats", tc.collStats}}}
+
+					targetCursor, targetErr := targetCollection.Aggregate(ctx, command)
+					compatCursor, compatErr := compatCollection.Aggregate(ctx, command)
+
+					if targetCursor != nil {
+						defer targetCursor.Close(ctx)
+					}
+					if compatCursor != nil {
+						defer compatCursor.Close(ctx)
+					}
+
+					if targetErr != nil {
+						t.Logf("Target error: %v", targetErr)
+						t.Logf("Compat error: %v", compatErr)
+						AssertMatchesCommandError(t, compatErr, targetErr)
+
+						return
+					}
+					require.NoError(t, compatErr, "compat error; target returned no error")
+
+					var targetRes, compatRes []bson.D
+					require.NoError(t, targetCursor.All(ctx, &targetRes))
+					require.NoError(t, compatCursor.All(ctx, &compatRes))
+
+					// $collStats returns one document per shard.
+					require.Equal(t, 1, len(compatRes))
+					require.Equal(t, 1, len(targetRes))
+
+					// Check the keys are the same
+					targetKeys := CollectKeys(t, targetRes[0])
+					compatKeys := CollectKeys(t, compatRes[0])
+
+					require.Equal(t, compatKeys, targetKeys)
```

Comment:
At the moment, we only check the key like `count`, `storageStats` exists. But not the values of the key. I think it's okay for now.